### PR TITLE
Event Activity Listing and Create

### LINKED ISSUE #38: Event Activity Listing Page
Page Url: /entity/event-activity/event/{eventId}
Module: Event - Activity
Authority
- CC Head
- Administrator
- Event Head
- Event Crew

Usage
View List of Event Activity as a timeline for planning

- [x] Development

### LINKED ISSUE #39: Create Event Activity Page
Page Url: /entity/event-activity/event/{eventId}/add
Module: Event - Activity
Authority
- CC Head
- Administrator
- Event Head
- Event Crew

Usage
Add event activity to event for planning

- [x] Development

### LINKED ISSUE #40: Update Event Activity Page
Page Url: /entity/event-activity/event/{eventId}/{id}/edit

Module: Event - Activity
Authority
- CC Head
- Administrator
- Event Head
- Event Crew

Usage
Update Event Activity for managing event activities/timeline

- [x] Development

### LINKED ISSUE #41: Delete Event Activity
Page Url: /entity/event-activity/event/{eventId}/{id}/delete
Module: Event - Activity
Authority
- CC Head
- Administrator
- Event Head
- Event Crew

Usage
Delete Event Activity from event

- [x] Development

### PR DESCRIPTION
**Pull request by HowardYaw**
- resolved #38 
- resolved #39 
- resolved #40 
- resolved #41

### CODE REVIEW

File: src/main/webapp/app/entities/event-activity/event-activity-create.tsx [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+export interface IEventActivityCreateState {
+  isNew: boolean;
+}
+
+export class EventActivityCreate extends React.Component<IEventActivityCreateProps, IEventActivityCreateState> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isNew: !this.props.match.params || !this.props.match.params.id
```
Comment (kachun333) [inline src/main/webapp/app/entities/event-activity/event-activity-create.tsx]: maybe this is cleaner?
```
!this.props.match?.params?.id
```

**Review by kachun333 — DISMISSED** [review]

File: src/main/webapp/app/entities/event-activity/event-activity-create.tsx (hunk 1 above)
Comment (HowardYaw) [reply src/main/webapp/app/entities/event-activity/event-activity-create.tsx]: i think i will remove this
since it is not using anymore 😆

Comment (HowardYaw) [reply src/main/webapp/app/entities/event-activity/event-activity-create.tsx]: resolved in 915d6223536e7bce48768a8e19037bd9a1cb9765

File: src/main/webapp/app/shared/util/tab.constants.ts [hunk 2]
```diff
@@ -11,6 +11,11 @@ export const eventTabList: ITabInfo[] = [
     tabTranslateKey: 'clubmanagementApp.eventChecklist.tab',
     tabRoute: '/event/checklist'
   },
+  {
+    tabName: 'Activities',
+    tabTranslateKey: 'clubmanagementApp.eventActivity.tab',
+    tabRoute: '/entity/event-activity/event/1'
```
Comment (kachun333) [inline src/main/webapp/app/shared/util/tab.constants.ts:40]: `/entity/event-activity/event/1`
are you sure this URL should be hardcoded?

**Review by kachun333 — DISMISSED** [review]

File: src/main/webapp/app/shared/util/tab.constants.ts (hunk 2 above)
Comment (HowardYaw) [reply src/main/webapp/app/shared/util/tab.constants.ts:40]: yes you are right this cannot be hardcoded
@kachun333 can I create a new issue and resolve this in another branch?
with all event module tab update

Comment (kachun333) [reply src/main/webapp/app/shared/util/tab.constants.ts:40]: @HowardYaw okay, no problem

File: src/main/webapp/i18n/zh-cn/eventActivity.json [hunk 3]
```diff
@@ -11,17 +13,17 @@
...
       },
       "eventId": "Event Id",
-      "startDate": "Start Date",
-      "durationInDay": "Duration In Day",
-      "name": "Name",
-      "description": "Description",
-      "tab": "Timeline"
+      "startDate": "开始日期",
+      "durationInDay": "为期（天数）",
+      "name": "名程",
```
Comment (LUXIANZE) [inline src/main/webapp/i18n/zh-cn/eventActivity.json]: typo

**Review by LUXIANZE — DISMISSED** [review]

File: src/main/webapp/i18n/zh-cn/eventActivity.json (hunk 3 above)
Comment (HowardYaw) [reply src/main/webapp/i18n/zh-cn/eventActivity.json]: resolved in dc6059809b4b12e952adc58bc61ae13b8c24de82

**Review by kachun333 — DISMISSED** [review]

File: src/main/webapp/app/entities/event-activity/event-activity-update.tsx [hunk 4]
```diff
@@ -126,31 +105,26 @@ export class EventActivityUpdate extends React.Component<IEventActivityUpdatePro
...
                   <Label id="descriptionLabel" for="event-activity-description">
                     <Translate contentKey="clubmanagementApp.eventActivity.description">Description</Translate>
                   </Label>
                   <AvInput id="event-activity-description" type="textarea" name="description" />
                 </AvGroup>
-                <Button tag={Link} id="cancel-save" to="/entity/event-activity" replace color="info">
-                  <FontAwesomeIcon icon="arrow-left" />
-                  &nbsp;
-                  <span className="d-none d-md-inline">
+                <span className="text-error">{errorMessage ? errorMessage.response.data.detail : ''}</span>
```
Comment (CarlosSia) [inline src/main/webapp/app/entities/event-activity/event-activity-update.tsx:114]: is this error message from startDate AvInput? it is not displayed when there is error

Comment (HowardYaw) [reply src/main/webapp/app/entities/event-activity/event-activity-update.tsx:114]: this is error message is show when get error from BE
the startDate Input will show below the input

File: src/main/webapp/app/entities/event-activity/event-activity.tsx [hunk 5]
```diff
@@ -53,98 +46,104 @@ export class EventActivity extends React.Component<IEventActivityProps, IEventAc
...
   render() {
-    const { eventActivityList, match, totalItems } = this.props;
+    const { eventActivityList, match, totalItems, selectedEventActivityId } = this.props;
     return (
       <div>
         <h2 id="event-activity-heading">
           <Translate contentKey="clubmanagementApp.eventActivity.home.title">Event Activities</Translate>
-          <Link to={`${match.url}/new`} className="btn btn-primary float-right jh-create-entity" id="jh-create-entity">
+        </h2>
+        <CustomTab tabList={eventTabList} currentTab="Activities" />
```
Comment (junzai97) [inline src/main/webapp/app/entities/event-activity/event-activity.tsx]: Shouldn't that the eventTabList have an argument of `eventId` due to the updated `tab.constants.ts`?

Comment (HowardYaw) [reply src/main/webapp/app/entities/event-activity/event-activity.tsx]: resolved in 948ee18a02eeba1d00248750fb41673c50daa860

**Review by junzai97 — APPROVED** [review]

**Review by CarlosSia — APPROVED** [review]

**Pull request merged**